File: translations.js
const fs = require('fs');
const path = require('path');

const translations = {
  de: require('./translations/de.json'),
  en: require('./translations/en.json'),
  he: require('./translations/he.json'),
  ja: require('./translations/ja.json'),
  ru: require('./translations/ru.json'),
  pt: require('./translations/pt.json'),
  es: require('./translations/es.json'),
  id: require('./translations/id.json'),
  zh: require('./translations/zh.json'),
  ar: require('./translations/ar.json')
};

const CONFIG_DIR = path.join(__dirname, 'configs');

function getGuildLanguage(guildId) {
  try {
    if (!guildId) return 'de';
    const configPath = path.join(CONFIG_DIR, `${guildId}.json`);

    if (!fs.existsSync(configPath)) return 'de';

    const config = JSON.parse(fs.readFileSync(configPath, 'utf8'));
    return config.language || 'de';
  } catch (error) {
    console.error('Error getting guild language:', error);
    return 'de';
  }
}

function setGuildLanguage(guildId, lang) {
  try {
    if (!guildId) return false;

    if (!['de', 'en', 'he', 'ja', 'ru', 'pt', 'es', 'id', 'zh', 'ar'].includes(lang)) {
      lang = 'de';
    }

    const configPath = path.join(CONFIG_DIR, `${guildId}.json`);
    let config = {};

    try {
      config = JSON.parse(fs.readFileSync(configPath, 'utf8'));
    } catch {}

    config.language = lang;
    fs.writeFileSync(configPath, JSON.stringify(config, null, 2));
    return true;
  } catch (error) {
    console.error('Error setting guild language:', error);
    return false;
  }
}

function t(guildId, key, replacements = {}) {
  const lang = getGuildLanguage(guildId);
  const keys = key.split('.');
  let value = translations[lang];

  for (const k of keys) {
    if (value && typeof value === 'object') {
      value = value[k];
    } else {
      return key;
    }
  }

  if (typeof value !== 'string') {
    return key;
  }

  let result = value;
  for (const [placeholder, replacement] of Object.entries(replacements)) {
    result = result.replace(new RegExp(`\\{${placeholder}\\}`, 'g'), replacement);
  }

  return result;
}

function getTranslations(lang = 'de') {
  return translations[lang] || translations.de;
}

function getLanguageName(lang) {
  const names = {
    de: 'Deutsch',
    en: 'English',
    he: 'עברית',
    ja: '日本語',
    ru: 'Русский',
    pt: 'Português',
    es: 'Español',
    id: 'Bahasa Indonesia',
    zh: '中文',
    ar: 'العربية'
  };
  return names[lang] || 'Deutsch';
}

module.exports = {
  getGuildLanguage,
  setGuildLanguage,
  t,
  getTranslations,
  getLanguageName
};

File: utils/translations.py
"""
Translation Module
Multi-language support for the bot
"""

import json
from pathlib import Path
from typing import Dict, Any, Optional

# Base directory
BASE_DIR = Path(__file__).parent.parent
TRANSLATIONS_DIR = BASE_DIR / "translations"
CONFIG_DIR = BASE_DIR / "configs"

# Load translations
translations: Dict[str, Dict[str, Any]] = {}

for lang_code in ['de', 'en', 'he']:
    lang_file = TRANSLATIONS_DIR / f"{lang_code}.json"
    if lang_file.exists():
        with open(lang_file, 'r', encoding='utf-8') as f:
            translations[lang_code] = json.load(f)


def get_guild_language(guild_id: Optional[str]) -> str:
    """
    Get guild's language from config.

    Args:
        guild_id: Discord guild ID

    Returns:
        Language code (de, en, or he)
    """
    try:
        if not guild_id:
            return 'de'  # Default to German

        config_path = CONFIG_DIR / f"{guild_id}.json"

        if not config_path.exists():
            return 'de'

        with open(config_path, 'r', encoding='utf-8') as f:
            config = json.load(f)
            return config.get('language', 'de')
    except Exception as e:
        print(f"Error getting guild language: {e}")
        return 'de'


def set_guild_language(guild_id: str, lang: str) -> bool:
    """
    Set guild's language in config.

    Args:
        guild_id: Discord guild ID
        lang: Language code (de, en, or he)

    Returns:
        True if successful, False otherwise
    """
    try:
        if not guild_id:
            return False

        # Validate language
        if lang not in ['de', 'en', 'he']:
            lang = 'de'

        config_path = CONFIG_DIR / f"{guild_id}.json"
        config = {}

        if config_path.exists():
            with open(config_path, 'r', encoding='utf-8') as f:
                config = json.load(f)

        config['language'] = lang

        with open(config_path, 'w', encoding='utf-8') as f:
            json.dump(config, f, indent=2, ensure_ascii=False)

        return True
    except Exception as e:
        print(f"Error setting guild language: {e}")
        return False


def t(guild_id: Optional[str], key: str, replacements: Optional[Dict[str, str]] = None) -> str:
    """
    Get translated text for a guild.

    Args:
        guild_id: Discord guild ID
        key: Translation key (e.g., 'ticket.created')
        replacements: Dictionary with placeholder replacements

    Returns:
        Translated text
    """
    if replacements is None:
        replacements = {}

    lang = get_guild_language(guild_id)
    keys = key.split('.')
    value = translations.get(lang, translations.get('de', {}))

    for k in keys:
        if isinstance(value, dict):
            value = value.get(k)
        else:
            return key  # Return key if translation not found

    if not isinstance(value, str):
        return key

    # Replace placeholders
    result = value
    for placeholder, replacement in replacements.items():
        result = result.replace(f'{{{placeholder}}}', str(replacement))

    return result


def get_translations(lang: str = 'de') -> Dict[str, Any]:
    """
    Get all translations for a language.

    Args:
        lang: Language code

    Returns:
        Translation dictionary
    """
    return translations.get(lang, translations.get('de', {}))


def get_language_name(lang: str) -> str:
    """
    Get language name.

    Args:
        lang: Language code

    Returns:
        Language name
    """
    names = {
        'de': 'Deutsch',
        'en': 'English',
        'he': 'עברית'
    }
    return names.get(lang, 'Deutsch')
